backend/src/v2/driver/k8s.go

```diff
@@ -482,6 +482,87 @@ func extendPodSpecPatch(
 		podSpec.Containers[0].VolumeMounts = append(podSpec.Containers[0].VolumeMounts, emptyDirVolumeMount)
 	}
 
+	// Get node affinity information
+	if nodeAffinityTerms := kubernetesExecutorConfig.GetNodeAffinity(); len(nodeAffinityTerms) > 0 {
+		var requiredTerms []k8score.NodeSelectorTerm
+		var preferredTerms []k8score.PreferredSchedulingTerm
+
+		for _, nodeAffinityTerm := range nodeAffinityTerms {
+			if nodeAffinityTerm.GetNodeAffinityJson() != nil {
+				var nodeAffinityJSON interface{}
+				err := resolveK8sJsonParameter(ctx, opts, dag, pipeline, mlmd,
+					nodeAffinityTerm.GetNodeAffinityJson(), inputParams, &nodeAffinityJSON)
+				if err != nil {
+					return fmt.Errorf("failed to resolve node affinity json: %w", err)
+				}
+
+				k8snodeAffinity, err := json.Marshal(nodeAffinityJSON)
+				if err != nil {
+					return fmt.Errorf("failed to marshal node affinity json: %w", err)
+				}
+				var nodeAffinity k8score.NodeAffinity
+				if err := json.Unmarshal(k8snodeAffinity, &nodeAffinity); err != nil {
+					return fmt.Errorf("failed to unmarshal node affinity json: %w", err)
+				}
```

Comment:
I may be missing something, but it doesn't seem right to me. Marshal and then unmarshal?

Reply:
Updated the code to avoid both marshal and then unmarshal.